### PR TITLE
fix(exercises): render DOM correctly with react

### PR DESCRIPTION
**Pull request by Tsugami**
errors when try run exercises 

```
export 'unstable_createRoot' (imported as 'createRoot') was not found in 'react-dom' (possible exports: __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED, createPortal, createRoot, findDOMNode, flushSync, hydrate, hydrateRoot, render, unmountComponentAtNode, unstable_batchedUpdates, unstable_renderSubtreeIntoContainer, version)
```

image

# Summary

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [ ] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Review by sibelius — COMMENTED** [review]
Can we use the new react18 createRoot method instead?

**Comment by Tsugami** [comment]
> react18 createRoot

we are not using the react 18 version, I think

autocomplete not found
image

**Comment by sibelius** [comment]
maybe you are in an old branch

check master https://github.com/sibelius/relay-workshop-private/blob/master/package.json#L88

**Comment by Tsugami** [comment]
> maybe you are in an old branch
> 
> check master https://github.com/sibelius/relay-workshop-private/blob/master/package.json#L88

I cloned the current master branch, maybe this specific version of 18 one doesn't have, I will test if is my vscode

**Comment by sibelius** [comment]
check this https://github.com/reactwg/react-18/discussions/5

try to do `yarn --force`

**Comment by Tsugami** [comment]
> yarn --force

image
image

interesting is that it causes more errors on the console as before

**Comment by Tsugami** [comment]
@sibelius can i try to update the version to the latest 18? do we have enough tests to not break anything?

**Comment by sibelius** [comment]
Please do

**Comment by Tsugami** [comment]
> Please do

I've tested the alpha and experimental version but same error :(

**Comment by sibelius** [comment]
@akinncar did this upgrade, try to talk to him

File: tsconfig.json [hunk 1]
```diff
@@ -49,6 +49,7 @@
     // "rootDirs": [],                        /* List of root folders whose combined content represents the structure of the project at runtime. */
     // "typeRoots": [],                       /* List of folders to include type definitions from. */
     // "types": [],                           /* Type declaration files to be included in compilation. */
+    "types": ["react/next", "react-dom/next"],
```
Comment (Tsugami) [inline tsconfig.json:52]: is ok in global tsconfig?

File: solutions/01-fetchGraphql/src/index.tsx [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 import React from 'react';
+import 'isomorphic-fetch';
 import ReactDOM from 'react-dom';
 
 import App from './App';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const container = document.getElementById('root');
+
+if (!container) throw new Error('Failed to find the root element');
```
Comment (sibelius) [inline solutions/01-fetchGraphql/src/index.tsx:9]: nice

**Review by sibelius — APPROVED** [review]
congrats

**Pull request merged**